common/hooks/do-extract/00-distfiles.sh

```diff
@@ -20,17 +20,16 @@ hook() {
 		fi
 	done
 
-	if [ -n "$create_wrksrc" ]; then
-		mkdir -p "${wrksrc}" || msg_error "$pkgver: failed to create wrksrc.\n"
-	fi
-
 	# Disable trap on ERR; the code is smart enough to report errors and abort.
 	trap - ERR
 
 	TAR_CMD="$(command -v bsdtar)"
 	[ -z "$TAR_CMD" ] && TAR_CMD="$(command -v tar)"
 	[ -z "$TAR_CMD" ] && msg_error "xbps-src: no suitable tar cmd (bsdtar, tar)\n"
 
+	extractdir=$(mktemp -d "$XBPS_BUILDDIR/.extractdir-XXXXXXX") ||
```

Comment:
should the tmpdir have the pkgname in it?

Reply:
Not really important, by why not :D

Reply:
I'm worry that `pkgname` may contain `XXX`, but no
> TEMPLATE must contain at least 3 consecutive 'X's in last component